src/DotNetty.Buffers/ByteBufferUtil.cs

```diff
@@ -569,5 +569,12 @@ static void AppendHexDumpRowPrefix(StringBuilder dump, int row, int rowStartInde
                 dump.Append('|');
             }
         }
+
+        public static string DecodeString(IByteBuffer src, int readerIndex, int len, Encoding encoding)
+        {
+            if (len == 0) return string.Empty;
+            
+            return encoding.GetString(src.Array.Slice(len));
```

Comment:
probably the easiest way to address this is to check if `buf.HasArray == false` and in this case, alloc new buffer from allocator and write necessary range of bytes there. It still implies that allocator should return buffer with `buf.HasArray == true` but I think it's OK to assume that. Now, if `buf.HasArray == true` you can use overload of GetString that decodes string from range of byte array: https://msdn.microsoft.com/en-us/library/05cts4c3(v=vs.110).aspx